### ISSUE TITLE
Using timestamp() in cypher causes Unknown function 'datetime' in the Gremlin layer

### ISSUE THREAD

**Issue by vinzenz** [ArcadeData/arcadedb#409]
### ArcadeDB Version: ArcadeDB Server v22.1.3-SNAPSHOT (build d38e26aa6978b19d0d69999919b5c076fc994d37/1642118408201/main)
Platform: Docker image from docker hub
JAVA_VERSION=jdk-11.0.13+8

## Expected behavior
No ERROR and successful call of timestamp()

## Actual behavior
ERROR with the following backtrace:

2022-05-24 20:29:59.546 WARNI [PostQueryHandler] <ArcadeDB_0> Found pending transaction from a previous operation. Rolling back it...<ArcadeDB_0> Error on command execution (PostCommandHandler)
com.arcadedb.exception.QueryParsingException: Error on executing Cypher query
	at com.arcadedb.gremlin.query.CypherQueryEngine.command(CypherQueryEngine.java:69)
	at com.arcadedb.database.EmbeddedDatabase.command(EmbeddedDatabase.java:1185)
	at com.arcadedb.server.ServerDatabase.command(ServerDatabase.java:395)
	at com.arcadedb.server.http.handler.PostCommandHandler.executeCommand(PostCommandHandler.java:238)
	at com.arcadedb.server.http.handler.PostCommandHandler.execute(PostCommandHandler.java:83)
	at com.arcadedb.server.http.handler.DatabaseAbstractHandler.execute(DatabaseAbstractHandler.java:82)
	at com.arcadedb.server.http.handler.AbstractHandler.handleRequest(AbstractHandler.java:108)
	at io.undertow.server.RoutingHandler.handleRequest(RoutingHandler.java:93)
	at io.undertow.server.handlers.PathHandler.handleRequest(PathHandler.java:104)
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:387)
	at io.undertow.server.protocol.http.HttpReadListener.handleEventWithNoRunningRequest(HttpReadListener.java:256)
	at io.undertow.server.protocol.http.HttpReadListener.handleEvent(HttpReadListener.java:136)
	at io.undertow.server.protocol.http.HttpReadListener.handleEvent(HttpReadListener.java:59)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at org.xnio.conduits.ReadReadyHandler$ChannelListenerHandler.readReady(ReadReadyHandler.java:66)
	at org.xnio.nio.NioSocketConduit.handleReady(NioSocketConduit.java:89)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:591)
Caused by: org.opencypher.gremlin.translation.exception.SyntaxException: Unknown function 'datetime'
	at org.opencypher.gremlin.translation.walker.ExpressionWalker.org$opencypher$gremlin$translation$walker$ExpressionWalker$$walkLocal(ExpressionWalker.scala:257)
	at org.opencypher.gremlin.translation.walker.ExpressionWalker$.walkLocal(ExpressionWalker.scala:49)
	at org.opencypher.gremlin.translation.walker.NodeUtils$.$anonfun$asList$2(NodeUtils.scala:167)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:233)
	at scala.collection.immutable.List.foreach(List.scala:388)
	at scala.collection.TraversableLike.map(TraversableLike.scala:233)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:226)
	at scala.collection.immutable.List.map(List.scala:294)
	at org.opencypher.gremlin.translation.walker.NodeUtils$.asList(NodeUtils.scala:167)
	at org.opencypher.gremlin.translation.walker.ExpressionWalker.asList(ExpressionWalker.scala:356)
	at org.opencypher.gremlin.translation.walker.ExpressionWalker.$anonfun$walkLocal$5(ExpressionWalker.scala:93)
	at scala.Option.getOrElse(Option.scala:121)
	at org.opencypher.gremlin.translation.walker.ExpressionWalker.org$opencypher$gremlin$translation$walker$ExpressionWalker$$walkLocal(ExpressionWalker.scala:91)
	at org.opencypher.gremlin.translation.walker.ExpressionWalker.walkLocal(ExpressionWalker.scala:72)
	at org.opencypher.gremlin.translation.walker.ExpressionWalker.walkProperty(ExpressionWalker.scala:331)
	at org.opencypher.gremlin.translation.walker.ExpressionWalker$.walkProperty(ExpressionWalker.scala:58)
	at org.opencypher.gremlin.translation.walker.SetWalker.setProperty(SetWalker.scala:113)
	at org.opencypher.gremlin.translation.walker.SetWalker.$anonfun$walkSetClause$1(SetWalker.scala:58)
	at scala.collection.immutable.List.foreach(List.scala:388)
	at org.opencypher.gremlin.translation.walker.SetWalker.walkSetClause(SetWalker.scala:56)
	at org.opencypher.gremlin.translation.walker.SetWalker.walkClause(SetWalker.scala:45)
	at org.opencypher.gremlin.translation.walker.SetWalker$.walkClause(SetWalker.scala:35)
	at org.opencypher.gremlin.translation.walker.MergeWalker.$anonfun$walkMerge$1(MergeWalker.scala:54)
	at org.opencypher.gremlin.translation.walker.MergeWalker.$anonfun$walkMerge$1$adapted(MergeWalker.scala:52)
	at scala.collection.immutable.List.foreach(List.scala:388)
	at org.opencypher.gremlin.translation.walker.MergeWalker.walkMerge(MergeWalker.scala:52)
	at org.opencypher.gremlin.translation.walker.MergeWalker.walkClause(MergeWalker.scala:39)
	at org.opencypher.gremlin.translation.walker.MergeWalker$.walkClause(MergeWalker.scala:31)
	at org.opencypher.gremlin.translation.walker.StatementWalker.walkClause(StatementWalker.scala:117)
	at org.opencypher.gremlin.translation.walker.StatementWalker.$anonfun$walkSingle$1(StatementWalker.scala:84)
	at org.opencypher.gremlin.translation.walker.StatementWalker.$anonfun$walkSingle$1$adapted(StatementWalker.scala:84)
	at scala.collection.immutable.List.foreach(List.scala:388)
	at org.opencypher.gremlin.translation.walker.StatementWalker.walkSingle(StatementWalker.scala:84)
	at org.opencypher.gremlin.translation.walker.StatementWalker.walk(StatementWalker.scala:45)
	at org.opencypher.gremlin.translation.walker.StatementWalker$.walk(StatementWalker.scala:32)
	at org.opencypher.gremlin.translation.CypherAst.translate(CypherAst.scala:98)
	at org.opencypher.gremlin.translation.CypherAst.buildTranslation(CypherAst.scala:119)
	at org.opencypher.gremlin.translation.TranslationFacade.toGremlinGroovy(TranslationFacade.java:56)
	at org.apache.tinkerpop.gremlin.arcadedb.structure.ArcadeCypher.compileToGremlin(ArcadeCypher.java:102)
	at org.apache.tinkerpop.gremlin.arcadedb.structure.ArcadeCypher.<init>(ArcadeCypher.java:53)
	at org.apache.tinkerpop.gremlin.arcadedb.structure.ArcadeGraph.cypher(ArcadeGraph.java:137)
	at com.arcadedb.gremlin.query.CypherQueryEngine.command(CypherQueryEngine.java:64)
	... 16 more

## Steps to reproduce
{cypher} MERGE (n:dog {value: 2}) ON CREATE SET n.created = timestamp(), n.updated = timestamp() ON MATCH SET n.updated = timestamp() RETURN n

**Comment by lvca** [comment]
I guess `timestamp()` function is not supported by Gremlin-Cypher project. I see there is a more active fork with support for date functions: https://github.com/chrisndodge/cypher-for-gremlin/commits/samos. I'm going to contact the author to understand what's the status of his branch.

**Comment by vinzenz** [comment]
So yeah there seems like it is, like you said.

For me this doesn't really hurry, I was evaluating several graph databases if they'd fit my goals and I came to the conclusion that I will have to go for relational design at this point. Even though it could have greatly benefited from a graph database.
However my requirements for the software kind of left me with Apache AGE due to memory restrictions I have to work with at this point. JAVA based systems are unfortunately too memory hungry. 
However AGE is way to immature at this point, I managed to get it crashed with some bigger queries, which I kind of needed for avoiding unnecessary round trips, however that wasn't even properly possible because of missing merge actions in AGE and Agensgraph just died :D 

I kind of like the CYPHER syntax, it's the most flexible for my needs and since it can be mixed together with SQL it's pretty powerful.

That said, I do highly appreciate all the work you are doing here and I definitely will keep ArcadeDB in mind on newer projects it looks like really an awesome piece of software however it doesn't fit my current restrictions.
